pkg/webhook/mutation.go

```diff
@@ -159,3 +174,43 @@ func AppendMutationWebhookIfEnabled(webhooks []rotator.WebhookInfo) []rotator.We
 	}
 	return webhooks
 }
+
+// validateGatekeeperResources returns whether an issue is user error (vs internal) and any errors
+// validating internal resources
+func (h *mutationHandler) validateGatekeeperResources(ctx context.Context, req admission.Request) (bool, error) {
+	if req.AdmissionRequest.Kind.Group == mutationsGroup && req.AdmissionRequest.Kind.Kind == "AssignMetadata" {
+		return h.validateAssignMetadata(ctx, req)
+	}
+	if req.AdmissionRequest.Kind.Group == mutationsGroup && req.AdmissionRequest.Kind.Kind == "Assign" {
+		return h.validateAssign(ctx, req)
+	}
+	return false, nil
+}
+
+func (h *mutationHandler) validateAssignMetadata(ctx context.Context, req admission.Request) (bool, error) {
+	obj, _, err := deserializer.Decode(req.AdmissionRequest.Object.Raw, nil, &mutationsv1alpha1.AssignMetadata{})
+	if err != nil {
+		return false, err
+	}
+	assignMetadata := obj.(*mutationsv1alpha1.AssignMetadata)
```

Comment:
we should do a safe type casting here (e.g. `assignMetadata, ok := obj.(*mutationsv1alpha1.AssignMetadata)`)